lib/rails_param/coercion/hash_param.rb

```diff
@@ -2,10 +2,14 @@ module RailsParam
   class Coercion
     class HashParam < VirtualParam
       def coerce
+        delimiter = options[:delimiter] || ","
+        separator = options[:separator] || ":"
+
         return param if param.is_a?(ActionController::Parameters)
         raise ArgumentError unless param.respond_to?(:split)
+        raise ArgumentError unless param.include?(separator)
 
-        Hash[param.split(options[:delimiter] || ",").map { |c| c.split(options[:separator] || ":") }]
+        Hash[param.split(delimiter).map { |c| c.split(separator) }]
```

Comment:
If you send a hash in the `arg1:val1,arg2:val2` format, this coercion worked correctly. It didn't correctly handle a situation when you would just send a string without a separator. If you would send `arg1` it would produce a hash as follows: `{ arg1: nil}` which I believe is an incorrect behaviour. There was a test testing for this, that was passing because of an error triggered elsewhere.